Userland/Libraries/LibWeb/Painting/PaintableBox.h

```diff
@@ -56,7 +56,11 @@ class PaintableBox : public Paintable {
         rect.set_x(absolute_rect.x() - box_model().padding.left);
         rect.set_width(content_width() + box_model().padding.left + box_model().padding.right);
         rect.set_y(absolute_rect.y() - box_model().padding.top);
-        rect.set_height(content_height() + box_model().padding.top + box_model().padding.bottom);
+        auto const& height = layout_node().computed_values().height();
+        if (height.is_auto())
+            rect.set_height(content_height() + box_model().padding.top + box_model().padding.bottom);
+        else
+            rect.set_height(content_height());
```

Comment:
Whatever the bug is, it should be fixed in layout, not in painting.
By the time we have a `PaintableBox`, all CSS *computed* values must have already been resolved to *used* values (i.e absolute coordinates, sizes, etc)

Reply:
See https://github.com/SerenityOS/serenity/blob/master/Documentation/Browser/LibWebFromLoadingToPainting.md for a brief introduction to the high-level phases of loading, parsing, layout, paint, etc. :)

Reply:
Making an adjustment in `FormattingContext::compute_height_for_absolutely_positioned_non_replaced_element` inspired from `BlockFormattingContext::compute_height` fixes the problem.

Should it be further aligned with `BlockFormattingContext::compute_height`? I'm happy to do it if that's desirable.

Reply:
Which part of which CSS spec does your fix implement? Can you link to the specific section?

Reply:
IIUC https://www.w3.org/TR/css-sizing-3/#box-sizing implies that the specified height includes padding, but it doesn't for `content-box`. I'll adjust further.

Reply:
Sure, I've updated the patch to do that as well.